client/dashboard/sites/deployments-list/branch-display.tsx

```diff
@@ -13,8 +13,6 @@ export function BranchDisplay( { branchName, color = '#3b3b3b' }: BranchDisplayP
 			<Text
 				as="code"
 				size="small"
-				truncate
```

Comment:
Removed truncation. Otherwise, the branch name will always be truncated to let the title field to get the most width. See the discussion here: https://github.com/Automattic/wp-calypso/pull/107236#discussion_r2545182272